src/tunnel.cc

```diff
@@ -1016,8 +1067,11 @@ void
 TunnelStateData::connectedToPeer(Security::EncryptorAnswer &answer)
 {
     if (ErrorState *error = answer.error.get()) {
-        answer.error.clear(); // sendError() will own the error
-        sendError(error, "TLS peer connection error");
+        saveError(error);
+        answer.error.clear();
+        if (CachePeer *p = server.conn->getPeer())
+            peerConnectFailed(p);
```

Comment:
Can this logic be moved to Security::BlindPeerConnector() or somesuch so that the callers do not need to remember to do this?

Reply:
> if BlindPeerConnector is created, peerConnectFailed() will be called twice.

This problem seems unrelated to this PR. I would add an XXX to FwdState::connectedToPeer(). Here is a sketch:

```
// XXX: Required for PeekingPeerConnector but duped by BlindPeerConnector.
```